app/models/subject_group.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SubjectGroup < ActiveRecord::Base
+  belongs_to :project
+
+  has_many :members, class_name: 'SubjectGroupMember', dependent: :destroy
+  has_many :subjects, through: :members
+  # a 'group' subject to record the grouped view of linked subject locations
+  # for use in talk and to collect retirement information about the group
+  belongs_to :group_subject, class_name: 'Subject'
+
+  validates :project, presence: true
+  validates :members, presence: true
+  validates :key, presence: true
```

Comment:
If it was me I'd probably add a uniqueness validation as well on `key` to prevent us hitting the error on the db.

Reply:
I want to avoid extra db queries on the model save / validation tbh - my plan was to use a find by key to ensure we reuse existing subject groups outside the model / create operation. 

If we need this extra check on validation, due to race conditions when creating subject groups, i'll add the key uniq validation step here.